src/snowflake/connector/connection_diagnostic.py

```diff
@@ -0,0 +1,578 @@
+                    self.__https_host_report(host, port=host_port, host_type=host_type)
+                except Exception:
+                    pass
+
+    def __is_privatelink(self) -> bool:
+        return "privatelink" in self.host
+
+    def __list_ips(
+        self,
+        host: str,
+        host_type: str = "SNOWFLAKE_URL"
+    ) -> None:
+        try:
+            ips = socket.gethostbyname_ex(host)[2]
+            base_message = f"{host}: nslookup results"
+
+
+            if "snowflakecomputing" in host:
+                for ip in ips:
+                    if ipaddress.ip_address(ip).is_private:
+                        if not self.__is_privatelink():
+                            self.__append_message(host_type,
+                                                  f"{base_message}: private ip: {ip}: WARNING: this is not "
+                                                  f"typical for a non-privatelink account")
+                    else:
+                         if self.__is_privatelink():
+                            self.__append_message(host_type,
+                                                  f"{base_message}: public ip: {ip}: WARNING: privatelink accounts "
+                                                  f"must have a private ip.")
+                         else:
+                            self.__append_message(host_type, f"{base_message}: public ip: {ip}")
+            else:
+                self.__append_message(host_type, f"{base_message}: {ips}")
+        except Exception as e:
+            logger.warning(f"Connectivity Test Exception in list_ips: {e}")
+
+    def __https_host_report(
+        self,
+        host: str,
+        port: int = 443,
+        host_type: str = "SNOWFLAKE_URL"
+    ):
+        try:
+            certificate = self.__test_socket_get_cert(host, port=port, host_type=host_type)
+            if "BEGIN CERTIFICATE" in certificate:
+                x509 = OpenSSL.crypto.load_certificate(OpenSSL.crypto.FILETYPE_PEM, certificate)
+
+                result = {
+                    'subject': dict(x509.get_subject().get_components()),
+                    'issuer': dict(x509.get_issuer().get_components()),
+                    'serialNumber': x509.get_serial_number(),
+                    'version': x509.get_version(),
+                    'notBefore': datetime.strptime(str(x509.get_notBefore().decode("utf-8")), '%Y%m%d%H%M%SZ'),
+                    'notAfter': datetime.strptime(str(x509.get_notAfter().decode("utf-8")), '%Y%m%d%H%M%SZ'),
+                }
+                self.cert_info[host] = result
+                extensions = (x509.get_extension(i) for i in range(x509.get_extension_count()))
+                extension_data = {}
+                for e in extensions:
+                    extension_data[e.get_short_name().decode("utf-8")] = str(e)
+
+                host_suffix = host.split('.')
+                host_suffix.pop(0)
+                host_suffix = '.'.join(host_suffix)
```

Comment:
While this does the same thing, just wanted to point out this other function that'd do the same thing without the need to reconstruct the string.
```suggestion
                _, _, host_suffix = host.partition('.')
```